src/Dapr.Actors/Client/ActorProxy.cs

```diff
@@ -305,5 +314,15 @@ protected bool CheckIfItsWrappedRequest(IActorRequestMessageBody requestMessageB
 
             return false;
         }
+
+        /// <summary>
+        /// Sets the Dapr Api Token which will be added to the header in all requests
+        /// </summary>
+        /// <param name="apiToken">Dapr Api Token value.</param>
+        public void SetDaprApiToken(string apiToken)
+        {
+            ArgumentVerifier.ThrowIfNull(apiToken, nameof(apiToken));
+            Environment.SetEnvironmentVariable("DAPR_API_TOKEN", apiToken);
```

Comment:
We shouldn't set environment variables for this process - this needs to get passed through to the right location in our code so that it's more flexible than an env-var. The user already has the ability to set an env-var without us providing an API :)